src/Module/FollowConfirm.php

```diff
@@ -0,0 +1,93 @@
+<?php
+namespace Friendica\Module;
+
+use Friendica\App;
+use Friendica\BaseModule;
+use Friendica\Core\L10n;
+use Friendica\Core\Logger;
+use Friendica\Core\Protocol;
+use Friendica\Database\DBA;
+use Friendica\Model\Contact;
+use Friendica\Protocol\Diaspora;
+use Friendica\Protocol\ActivityPub;
+use Friendica\Util\DateTimeFormat;
+
+class FollowConfirm extends BaseModule
+{
+	public static function post()
+	{
+		$a = self::getApp();
+
+		$uid = local_user();
+		if (!$uid) {
+			notice(L10n::t('Permission denied.') . EOL);
+			return;
+		}
+
+		$intro_id = intval(defaults($_POST, 'intro_id'  , 0));
+		$duplex   = intval(defaults($_POST, 'duplex'    , 0));
+		$cid      = intval(defaults($_POST, 'contact_id', 0));
+		$hidden   = intval(defaults($_POST, 'hidden'    , 0));
+
+		if (empty($cid)) {
+			notice(L10n::t('No given contact.') . EOL);
+			return;
+		}
+
+		Logger::log('Confirming follower with contact_id: ' . $cid);
+
+		$contact = DBA::selectFirst('contact', [], ['id' => $cid, 'uid' => $uid]);
+		if (!DBA::isResult($contact)) {
+			Logger::log('Contact not found in DB.');
```

Comment:
```suggestion
			$a->getLogger()->info('Contact not found in DB.', ['cid' => $cid, 'uid' => $uid]);
```

Reply:
Why ```$a->getLogger()->``` instead of ```Logger::```?

Reply:
It just retrieves the Logger object directly instead of trusting the Logger static class.

Reply:
I just find this more complicated.

Reply:
The `Logger` static class just hides the complexity, making it harder to troubleshoot in case of an issue arising from this hidden complexity, remember the nasty infinite loop between `DBA`, `Logger` and `Config`. All those classes are super convenient, and yet we got severely blind-sided.

You're right about one thing about complexity, though, the ideal call should be
```php
$this->logger->info(...);
```
So the work isn't done yet in that regard.

Reply:
this would be the case if we use the DI-library:
```php
class FollowConfirm
{
    /**
      * @var LoggerInterface
     **/
    private $logger;

    public function __construct(LoggerInterface $logger)
    {
        $this->logger = $logger;
    }

    public function post()
    {
        $this->logger->info(...);
    }
}
```

and the call for `App\Router` would be:
```php
class Router
{
    public function getModule()
    {
        return $this->dice->create(FollowConfirm::class);
    }
}
```
which automatically fill the `LoggerInterface $logger` either with the "real" dependency, or - in case of tests - just with the `VoidLogger` :-)